src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorDimensionSpan_1.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Numerics.Tensors
+{
+    /// <summary>Represents the slices that exist within a dimension of a tensor span.</summary>
+    /// <typeparam name="T">The type of the elements within the tensor span.</typeparam>
+    public readonly ref struct ReadOnlyTensorDimensionSpan<T>
+    {
+        private readonly ReadOnlyTensorSpan<T> _tensor;
+        private readonly nint _length;
+        private readonly int _dimension;
+        private readonly TensorShape _sliceShape;
+
+        internal ReadOnlyTensorDimensionSpan(ReadOnlyTensorSpan<T> tensor, int dimension)
+        {
+            if ((uint)dimension >= tensor.Rank)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+            }
+            dimension += 1;
+
+            _tensor = tensor;
+            _length = TensorPrimitives.Product(tensor.Lengths[..dimension]);
+            _dimension = dimension;
+            _sliceShape = TensorShape.Create((dimension != tensor.Rank) ? tensor.Lengths[dimension..] : [1], tensor.Strides[dimension..]);
+        }
+
+        /// <summary>Gets the length of the tensor dimension span.</summary>
+        public nint Length => _length;
+
+        /// <summary>Gets the tensor span representing a slice of the tracked dimension using the specified index.</summary>
+        /// <param name="index">The index of the tensor span slice to retrieve within the tracked dimension.</param>
+        /// <returns>The tensor span representing a slice of the tracked dimension using <paramref name="index" />.</returns>
+        public ReadOnlyTensorSpan<T> this[nint index]
+        {
+            get
+            {
+                if ((nuint)index >= (nuint)_length)
+                {
+                    ThrowHelper.ThrowArgumentOutOfRangeException();
+                }
+
+                nint linearOffset = _tensor._shape.GetLinearOffset(index, _dimension);
+                return new ReadOnlyTensorSpan<T>(ref Unsafe.Add(ref _tensor._reference, linearOffset), _sliceShape);
+            }
+        }
+
+        /// <summary>Gets an enumerator for the readonly tensor dimension span.</summary>
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>Enumerates the spans of a tensor dimension span.</summary>
+        public ref struct Enumerator
+#if NET9_0_OR_GREATER
+            : IEnumerator<ReadOnlyTensorSpan<T>>
```

Comment:
Should this implement other interfaces in older TFMs?

Reply:
What other interfaces would it implement?